doc/design/embargo-1.md

```diff
@@ -0,0 +1,95 @@
+# Embargo support
+
+## Principles
+
```

Comment:
When a user creates a dandiset, they have the option to mark it as "public" or "embargoed". There is some help text that explains these options, either as a hover-hint or some other way. The text for "public" would indicate that the dandiset would be visible to anyone. 

If "embargoed" is selected, a page would be displayed that would explain the rules of embargoed data. The user is required to check "I agree" before proceeding. The message would indicate that this option is only available to BRAIN Initiative grants. The landing page of the dandiset would be visible to everyone but the underlying data would not be. As per recent NIH policy, the user will be required to open the dandiset upon the first publication of the data, or when the funding for the grant runs out, whichever comes first. The user would be required to enter the grant or award number, and we would ideally validate that this is an active award. We will be able to determine when the grant expires and we will automatically open the dandiset if it has not already been opened. It is expected that each grant have at least one of these embargoed dandisets. Since it is not always clear from the outset how the data from a grant will be divided into publications, it is expected that several groups may start with a single dandiset.

For embargoed data the owner would have options to add user permissions, with an interface similar to Google Docs:
* add read-only. This would be used for collaborations with team members who do not need to submit data but only to read it. This would also be useful for reviewers of manuscripts who need to see the data as part of the review process. Read access extends to downloading data with the python API, and reading data directly using s3 streaming. A DANDI Hub user will automatically use the credentials from the dandi hub account they are currently using.
* add collaborator. This would give them write permissions
* add owner. This would give them read and write permissions and the ability to add users, including owners.

Permissions are all my dandiset, not by asset. If a user needs to set different permissions for different assets within the dandiset, they will need to create a new dandiset and put the assets in this separate place.

When a there is a new publication, the user is expected to create a new dandiset for this publication. There should be an online interface for copying assets from an existing dandiset to this one.

Reply:
Thank you @bendichter !

Re use of "public". To some degree, with making those dandiset-level metadata available, they already become "public" under an obscure notion of "public" many archives use -- people could see them and request access to their data.  That is why I followed the terminology we have in the dandischema already, and called them `"open"`.

The other notes -- might better reside in more specific sections of the design (e.g. discussion of the web ui interactions) . 

"Permissions are all by dandiset, not by asset" - is a great one to go here in these high level principles.

Great point on separating "owner" from "collaborator", but somewhat orthogonal to "embargo" IMHO. ATM I believe we have no "owner"-vs-"collaborator" distinction. We could mix it in here, or postpone (to concentrate on embargo aspects) for a separate discussion/implementation.